pkg/data/domain/cluster/aws.go

```diff
@@ -0,0 +1,94 @@
+package cluster
+
+import (
+	"context"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	capiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (s *Service) getAllAWS(ctx context.Context, namespace string) (*infrastructurev1alpha2.AWSClusterList, error) {
+	var err error
+
+	options := &runtimeClient.ListOptions{
+		Namespace: namespace,
+	}
+
+	var clusterIDs map[string]bool
+	{
+		// The CAPI clusters have the labels. They can be used for filtering.
+		apiClusters := &capiv1alpha2.ClusterList{}
```

Comment:
Getting Cluster API clusters first, since this have labels, and can be used for filtering